src/test/java/org/broadinstitute/hellbender/tools/walkers/annotator/VariantAnnotatorIntegrationTest.java

```diff
@@ -0,0 +1,392 @@
+//        executeTest("Testing InbreedingCoeff annotation with PED file", spec);
+//    }
+
+    @Test
+    public void testAlleleTrimming() throws IOException {
+        // This test makes sure that the expression code works in a complex case with many overlapping variant contexts
+        assertVariantContextsMatch(getTestFile("AlleleTrim.vcf"),
+                getTestFile("expected/testAlleleTrimming.vcf"),
+                Arrays.asList( "--resource", "exac:" + getToolTestDataDir() + "exacAlleleTrim.vcf", "-E", "exac.AC_Adj", "-A", "InbreedingCoeff"),
+                null, Collections.emptyList());
+    }
+
+    @Test
+    public void testStrandBiasBySample() throws IOException {
+        // Created variants via HalotypeCaller GATK3 with no default annotations
+        final File outputVCF = getTestFile("HCOutputNoAnnotations.vcf");
+
+        // Created variant via HalotypeCaller GATK3; include StrandBiasBySample, exclude FisherStrand annotation
+        //             re-Annotate the variant with VariantAnnotator using FisherStrand annotation
+        final File outputVCFNoFS = getTestFile("HCOutputNoFSAnnotation.vcf");
+
+        final File outputWithAddedFS = createTempFile("variantannotator", ".vcf");
+
+        final ArgumentsBuilder args = new ArgumentsBuilder();
+        args.addReference(new File(b37_reference_20_21));
+        args.addVCF(outputVCFNoFS);
+        args.addOutput(outputWithAddedFS);
+        args.add("-L 20:10130000-10134800");
+        args.add("-A FisherStrand");
+        runCommandLine(args);
+
+        // confirm that the FisherStrand values are identical for the two pipelines
+        final VCFCodec codec = new VCFCodec();
+        final FileInputStream s = new FileInputStream(outputVCF);
+        final LineIterator lineIterator = codec.makeSourceFromStream(new PositionalBufferedStream(s));
+        codec.readHeader(lineIterator);
+
+        final VCFCodec codecAnn = new VCFCodec();
+        final FileInputStream sAnn = new FileInputStream(outputWithAddedFS);
+        final LineIterator lineIteratorAnn = codecAnn.makeSourceFromStream(new PositionalBufferedStream(sAnn));
+        codecAnn.readHeader(lineIteratorAnn);
+
+        while( lineIterator.hasNext() && lineIteratorAnn.hasNext() ) {
+            final String line = lineIterator.next();
+            Assert.assertFalse(line == null);
+            final VariantContext vc = codec.decode(line);
+
+            final String lineAnn = lineIteratorAnn.next();
+            Assert.assertFalse(lineAnn == null);
+            final VariantContext vcAnn = codecAnn.decode(lineAnn);
+
+            Assert.assertTrue(vc.hasAttribute("FS"));
+            Assert.assertTrue(vcAnn.hasAttribute("FS"));
+            Assert.assertEquals(vc.getAttributeAsDouble("FS", 0.0), vcAnn.getAttributeAsDouble("FS", -1.0));
+        }
+
+        Assert.assertFalse(lineIterator.hasNext());
+        Assert.assertFalse(lineIteratorAnn.hasNext());
+    }
+
+}
```

Comment:
There's nothing specific that jumps out at me as being missing.  That being said, in GATK3, this was pretty much the only way we got test coverage on most of the annotations.  In GATK4 there are a lot more unit tests, so maybe that's less of an issue.  How hard would it be to run test coverage over the classes in the walkers.annotator package?  I knew how to do it in IntelliJ once, but it's been a while.

Reply:
I have run coverage, there are a few annotations that could use better coverage. Notable among them are StrandArtifact, RMSMappingQuality (though that seems to be a fluke because it does have tests), OxoGCounts, MappingQuality, Fragment Length, and DepthPerAlleleBySample. I have added a test for --use-all-annotationsthat asserts everything at least ran and is outputting results if I think it should but it's by no means comprehensive (just testing for crashes)

Reply:
Of which I have found a few in our seldom-used annotations, namely StrandArtifact and ReferenceBases. These have been resolved now.

Reply:
OxoGCounts is probably going to be obsolete soon.  Not sure if @davidbenjamin still cares about MappingQuality (median MQ per allele).  DepthPerAlleleBySample is AD, so that should be tested in a lot of places -- maybe that's a fluke too.  I think annotation coverage is good enough as is.

Reply:
MappingQuality is a default annotation of Mutect2 and remains useful to us.